docs/design/AnoncredsW3CCompatibility.md

```diff
@@ -0,0 +1,286 @@
+
+This design proposes to extend the Aries Cloud Agent Python (ACA-PY) to support Hyperledger AnonCreds credentials and presentations in the W3C Verifiable Credentials (VC) and Verifiable Presentations (VP) Format. The aim is to transition from the legacy AnonCreds format specified in Aries-Legacy-Method to the W3C VC format.
+<br><br>
+
+## Overview
+
+We aim to wrap the enhancements made on the Rust Framework for Anoncreds first. Then we will work on the integration of AnonCreds with W3C VC Format in ACA-PY, which includes support for issuing, verifying, and managing W3C VC Format AnonCreds credentials.
+
+Ideally the signatures will be delivered in parallel with the Javascript Framework Document.
+<br><br>
+
+## Caveats
+
+For now, we will only target compatibility with [VCDM (Verifiable Credential Data Model) 1.1](https://www.w3.org/TR/vc-data-model/) because the Rust framework we are deriving from is also working with that target. This means we would support the features being implemented in the RUST frameworks, which include:
+
+- Credentials: Verify validity of non-Creds Data Integrity proof signatures
+- Presentations: Create presentations using non-AnonCreds Data Integrity proof signature
+- Presentations: Verify validity of presentations, including non-AnonCreds Data Integrity proof signatures
+- Presentations: Support different formats (for example, DIF) of Presentation Request
+
+This may change if the Rust framework decides to support VCDM (Verifiable Credential Data Model) 2.0.
+<br><br>
+
+## Issues to consider
+
+- If and how the W3C VC Format attachments for the Issue Credential V2.0 and Present Proof V2 Aries DIDComm Protocols should be used when using AnonCreds W3C VC Format credentials.
+- How AnonCreds W3C VC Format verifiable credentials are stored by the holder such that they will be discoverable when needed for creating verifiable presentations.
+- How and when multiple signatures can/should be added to a W3C VC Format credential, enabling both AnonCreds and non-AnonCreds signatures on a single credential and their use in presentations.
+  <br><br>
+
+## Flow Chart
+
+![image](./anoncreds-w3c-verification-flow.png)
+
+## Key Questions
+
+### What is the roadmap for delivery? What will we build first, then second?
+
+1. Python wrappers around Rust
+2. W3C VC conversion (`to_w3c()`) in ACA-PY.
+3. W3C VC issuance in ACA-PY.
+4. W3C VC storage in holder's wallet in ACA-PY.
+5. W3C VC presentation from an AnonCreds cred in ACA-PY.
+6. Convert W3C VCs back to AnonCreds (`to_legacy()`) in ACA-PY.
+
+### What are the functions we are going to wrap?
+
+After thoroughly reviewing upcoming changes from [anoncreds-rs PR273](https://github.com/hyperledger/anoncreds-rs/pull/273), the classes or `AnoncredsObject` that we deemed necessary to be exported are:<br>
+
+[W3CCredentialOffer](https://github.com/hyperledger/anoncreds-rs/pull/273/files#diff-6f8cbd34bbd373240b6af81f159177023c05b074b63c7757fc6b3796a66ee240R106)<br>
```

Comment:
Minor point -- per the last meeting -- the "W3C-specific" versions of these calls have been dropped as they are identical to the existing calls.

Reply:
Ok, I've checked that `W3cCredentialOffer` and `W3cCredentialRequest` are removed. will remove them as well.